packages/react/src/TextInput/TextInput.tsx

```diff
@@ -96,6 +100,21 @@ const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(
     const focusInput: MouseEventHandler = () => {
       inputRef.current?.focus()
     }
+    const leadingVisualId = useId()
+    const trailingVisualId = useId()
+    const loadingId = useId()
+
+    const inputDescribedBy =
+      [
+        inputProps['aria-describedby'],
+        LeadingVisual && leadingVisualId,
+        TrailingVisual && trailingVisualId,
+        loading && loadingId,
+      ]
+        .filter(String)
+        .join(' ')
+        .trim() || undefined
```

Comment:
We could use `clsx` for this. It was technically designed for classes but should work for any HTML attribute that has space-separated values.:

```suggestion
    const inputDescribedBy =
      clsx(
        inputProps['aria-describedby'],
        LeadingVisual && leadingVisualId,
        TrailingVisual && trailingVisualId,
        loading && loadingId,
      )
```